src/bp/core/services/stats/index.ts

```diff
@@ -0,0 +1,215 @@
+import axios from 'axios'
+import LicensingService from 'common/licensing-service'
+import { machineUUID } from 'common/stats'
+import { ConfigProvider } from 'core/config/config-loader'
+import Database from 'core/database'
+import { UserRepository } from 'core/repositories'
+import { TYPES } from 'core/types'
+import { inject, injectable } from 'inversify'
+import ms from 'ms'
+import os from 'os'
+import uuid from 'uuid'
+import yn from 'yn'
+
+import { GhostService } from '..'
+import AuthService from '../auth/auth-service'
+import { BotService } from '../bot-service'
+import { CMSService } from '../cms'
+import { JobService } from '../job-service'
+import { WorkspaceService } from '../workspace-service'
+
+const LOCK_RESOURCE = 'botpress:statsService'
+const debug = DEBUG('stats')
+
+@injectable()
+export class StatsService {
+  constructor(
+    @inject(TYPES.ConfigProvider) private config: ConfigProvider,
+    @inject(TYPES.JobService) private jobService: JobService,
+    @inject(TYPES.BotService) private botService: BotService,
+    @inject(TYPES.GhostService) private ghostService: GhostService,
+    @inject(TYPES.LicensingService) private licenseService: LicensingService,
+    @inject(TYPES.WorkspaceService) private workspaceService: WorkspaceService,
+    @inject(TYPES.CMSService) private cmsService: CMSService,
+    @inject(TYPES.AuthService) private authService: AuthService,
+    @inject(TYPES.UserRepository) private userRepository: UserRepository,
+    @inject(TYPES.Database) private database: Database
+  ) {}
+
+  public start() {
+    // tslint:disable-next-line: no-floating-promises
+    this.run()
+    setInterval(this.run.bind(this), ms('6 hours'))
+  }
+
+  private async run() {
+    const lock = await this.jobService.acquireLock(LOCK_RESOURCE, ms('6 hours') - ms('1 minute'))
```

Comment:
I suggest moving the 6 hours in a constant since it's used twice and it may have an impact if one is forgotten